TypeError: index() got an unexpected keyword argument 'headers' with sample code

I have written a `pytest` version of the sample code.

```python
import elasticsearch

class FooService:

    def __init__(self):
        self.es = elasticsearch.Elasticsearch(hosts=[{'host': 'localhost', 'port': 9200}])

    def create(self, index, body):
        es_object = self.es.index(index, body)
        return es_object.get('_id')

    def read(self, index, id):
        es_object = self.es.get(index, id)
        return es_object.get('_source')

import pytest
from elasticmock import elasticmock

@elasticmock
def test_create_and_read_object():
    # Variables used to test
    index = 'test-index'
    expected_document = {
        'foo': 'bar'
    }

    # Instantiate service
    service = FooService()

    # Index document on ElasticSearch
    id = service.create(index, expected_document)
    assert id is not None

    # Retrieve document from ElasticSearch
    document = service.read(index, id)
    assert expected_document == document
```

When I run it I get the following error

    $ pytest em.py 
    ========================================== test session starts ===========================================
    platform darwin -- Python 3.7.6, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
    rootdir: /Users/wmcneill/Documents/Misc
    plugins: celery-4.4.2
    collected 1 item                                                                                         

    em.py F                                                                                            [100%]

    ================================================ FAILURES ================================================
    ______________________________________ test_create_and_read_object _______________________________________

        @elasticmock
        def test_create_and_read_object():
            # Variables used to test
            index = 'test-index'
            expected_document = {
                'foo': 'bar'
            }
        
            # Instantiate service
            service = FooService()
        
            # Index document on ElasticSearch
    >       id = service.create(index, expected_document)

    em.py:31: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    em.py:9: in create
        es_object = self.es.index(index, body)
    ../../greenfield/document_store/venv/lib/python3.7/site-packages/elasticmock/behaviour/server_failure.py:27: in decorated
        response = f(*args, **kwargs)
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    args = (<elasticmock.fake_elasticsearch.FakeElasticsearch object at 0x1051cc110>, 'test-index', {'foo': 'bar'})
    kwargs = {}, params = {}, headers = {}, p = 'request_timeout'

        @wraps(func)
        def _wrapped(*args, **kwargs):
            params = {}
            headers = {}
            if "params" in kwargs:
                params = kwargs.pop("params").copy()
            if "headers" in kwargs:
                headers = {
                    k.lower(): v for k, v in (kwargs.pop("headers") or {}).items()
                }
            if "opaque_id" in kwargs:
                headers["x-opaque-id"] = kwargs.pop("opaque_id")
        
            for p in es_query_params + GLOBAL_PARAMS:
                if p in kwargs:
                    v = kwargs.pop(p)
                    if v is not None:
                        params[p] = _escape(v)
        
            # don't treat ignore, request_timeout, and opaque_id as other params to avoid escaping
            for p in ("ignore", "request_timeout"):
                if p in kwargs:
                    params[p] = kwargs.pop(p)
    >       return func(*args, params=params, headers=headers, **kwargs)
    E       TypeError: index() got an unexpected keyword argument 'headers'

    ../../greenfield/document_store/venv/lib/python3.7/site-packages/elasticsearch/client/utils.py:92: TypeError
    ======================================== short test summary info =========================================
    FAILED em.py::test_create_and_read_object - TypeError: index() got an unexpected keyword argument 'head...
    =========================================== 1 failed in 0.28s ============================================

This is ElasticMock version 1.5.0 and elasticsearch version 7.6.0.

--- a/elasticmock/fake_cluster.py
+++ b/elasticmock/fake_cluster.py
@@ -9,7 +9,7 @@ class FakeClusterClient(ClusterClient):
     @query_params('level', 'local', 'master_timeout', 'timeout',
                   'wait_for_active_shards', 'wait_for_nodes',
                   'wait_for_relocating_shards', 'wait_for_status')
-    def health(self, index=None, params=None):
+    def health(self, index=None, params=None, headers=None):
         return {
             'cluster_name': 'testcluster',
             'status': 'green',

--- a/elasticmock/fake_elasticsearch.py
+++ b/elasticmock/fake_elasticsearch.py
@@ -35,11 +35,11 @@ class FakeElasticsearch(Elasticsearch):
         return FakeClusterClient(self)
 
     @query_params()
-    def ping(self, params=None):
+    def ping(self, params=None, headers=None):
         return True
 
     @query_params()
-    def info(self, params=None):
+    def info(self, params=None, headers=None):
         return {
             'status': 200,
             'cluster_name': 'elasticmock',
@@ -57,7 +57,7 @@ class FakeElasticsearch(Elasticsearch):
 
     @query_params('consistency', 'op_type', 'parent', 'refresh', 'replication',
                   'routing', 'timeout', 'timestamp', 'ttl', 'version', 'version_type')
-    def index(self, index, body, doc_type='_doc', id=None, params=None):
+    def index(self, index, body, doc_type='_doc', id=None, params=None, headers=None):
         if index not in self.__documents_dict:
             self.__documents_dict[index] = list()
         
@@ -88,7 +88,7 @@ class FakeElasticsearch(Elasticsearch):
 
     @query_params('consistency', 'op_type', 'parent', 'refresh', 'replication',
                   'routing', 'timeout', 'timestamp', 'ttl', 'version', 'version_type')
-    def bulk(self,  body, index=None, doc_type=None, params=None):
+    def bulk(self,  body, index=None, doc_type=None, params=None, headers=None):
         version = 1
         items = []
 
@@ -128,7 +128,7 @@ class FakeElasticsearch(Elasticsearch):
         }
 
     @query_params('parent', 'preference', 'realtime', 'refresh', 'routing')
-    def exists(self, index, doc_type, id, params=None):
+    def exists(self, index, doc_type, id, params=None, headers=None):
         result = False
         if index in self.__documents_dict:
             for document in self.__documents_dict[index]:
@@ -140,7 +140,7 @@ class FakeElasticsearch(Elasticsearch):
     @query_params('_source', '_source_exclude', '_source_include', 'fields',
                   'parent', 'preference', 'realtime', 'refresh', 'routing', 'version',
                   'version_type')
-    def get(self, index, id, doc_type='_all', params=None):
+    def get(self, index, id, doc_type='_all', params=None, headers=None):
         result = None
         if index in self.__documents_dict:
             for document in self.__documents_dict[index]:
@@ -169,7 +169,7 @@ class FakeElasticsearch(Elasticsearch):
     @query_params('_source', '_source_exclude', '_source_include', 'parent',
                   'preference', 'realtime', 'refresh', 'routing', 'version',
                   'version_type')
-    def get_source(self, index, doc_type, id, params=None):
+    def get_source(self, index, doc_type, id, params=None, headers=None):
         document = self.get(index=index, doc_type=doc_type, id=id, params=params)
         return document.get('_source')
 
@@ -181,7 +181,7 @@ class FakeElasticsearch(Elasticsearch):
                   'size', 'sort', 'stats', 'suggest_field', 'suggest_mode',
                   'suggest_size', 'suggest_text', 'terminate_after', 'timeout',
                   'track_scores', 'version')
-    def count(self, index=None, doc_type=None, body=None, params=None):
+    def count(self, index=None, doc_type=None, body=None, params=None, headers=None):
         searchable_indexes = self._normalize_index_to_list(index)
 
         i = 0
@@ -209,7 +209,7 @@ class FakeElasticsearch(Elasticsearch):
                   'size', 'sort', 'stats', 'suggest_field', 'suggest_mode',
                   'suggest_size', 'suggest_text', 'terminate_after', 'timeout',
                   'track_scores', 'version')
-    def search(self, index=None, doc_type=None, body=None, params=None):
+    def search(self, index=None, doc_type=None, body=None, params=None, headers=None):
         searchable_indexes = self._normalize_index_to_list(index)
 
         matches = []
@@ -273,7 +273,7 @@ class FakeElasticsearch(Elasticsearch):
         return result
 
     @query_params('scroll')
-    def scroll(self, scroll_id, params=None):
+    def scroll(self, scroll_id, params=None, headers=None):
         scroll = self.__scrolls.pop(scroll_id)
         result = self.search(
             index=scroll.get('index'),
@@ -285,7 +285,7 @@ class FakeElasticsearch(Elasticsearch):
     
     @query_params('consistency', 'parent', 'refresh', 'replication', 'routing',
                   'timeout', 'version', 'version_type')
-    def delete(self, index, doc_type, id, params=None):
+    def delete(self, index, doc_type, id, params=None, headers=None):
 
         found = False
 
@@ -311,7 +311,7 @@ class FakeElasticsearch(Elasticsearch):
 
     @query_params('allow_no_indices', 'expand_wildcards', 'ignore_unavailable',
                   'preference', 'routing')
-    def suggest(self, body, index=None, params=None):
+    def suggest(self, body, index=None, params=None, headers=None):
         if index is not None and index not in self.__documents_dict:
             raise NotFoundError(404, 'IndexMissingException[[{0}] missing]'.format(index))
 

--- a/elasticmock/fake_indices.py
+++ b/elasticmock/fake_indices.py
@@ -7,23 +7,23 @@ from elasticsearch.client.utils import query_params
 class FakeIndicesClient(IndicesClient):
 
     @query_params('master_timeout', 'timeout')
-    def create(self, index, body=None, params=None):
+    def create(self, index, body=None, params=None, headers=None):
         documents_dict = self.__get_documents_dict()
         if index not in documents_dict:
             documents_dict[index] = []
 
     @query_params('allow_no_indices', 'expand_wildcards', 'ignore_unavailable',
                   'local')
-    def exists(self, index, params=None):
+    def exists(self, index, params=None, headers=None):
         return index in self.__get_documents_dict()
 
     @query_params('allow_no_indices', 'expand_wildcards', 'force',
                   'ignore_unavailable', 'operation_threading')
-    def refresh(self, index=None, params=None):
+    def refresh(self, index=None, params=None, headers=None):
         pass
 
     @query_params('master_timeout', 'timeout')
-    def delete(self, index, params=None):
+    def delete(self, index, params=None, headers=None):
         documents_dict = self.__get_documents_dict()
         if index in documents_dict:
             del documents_dict[index]

